tools/who_what_benchmark/whowhatbench/image2image.py

```diff
@@ -0,0 +1,143 @@
+import os
+from typing import Any, Union
+
+import datasets
+import pandas as pd
+from tqdm import tqdm
+from transformers import set_seed
+import torch
+import openvino_genai
+
+from .registry import register_evaluator
+from .text2image_evaluator import Text2ImageEvaluator
+
+from .whowhat_metrics import ImageSimilarity
+
+
+class Generator(openvino_genai.Generator):
+    def __init__(self, seed, rng, mu=0.0, sigma=1.0):
+        openvino_genai.Generator.__init__(self)
+        self.mu = mu
+        self.sigma = sigma
+        self.rng = rng
+
+    def next(self):
+        return torch.randn(1, generator=self.rng, dtype=torch.float32).item()
+
+
+def preprocess_fn(example):
+    return {
+        "prompts": example["Instruction_VLM-LLM"],
+        "images": example["source_img"],
+    }
+
+
+def prepare_default_data(num_samples=None):
+    DATASET_NAME = "paint-by-inpaint/PIPE"
+    NUM_SAMPLES = 10 if num_samples is None else num_samples
+    set_seed(42)
+    default_dataset = datasets.load_dataset(
+        DATASET_NAME, split="test", streaming=True
+    ).filter(lambda example: example["Instruction_VLM-LLM"] != "").take(NUM_SAMPLES)
+    return default_dataset.map(
+        lambda x: preprocess_fn(x), remove_columns=default_dataset.column_names
+    )
+
+
+@register_evaluator("image-to-image")
+class Image2ImageEvaluator(Text2ImageEvaluator):
+    def __init__(
+        self,
+        base_model: Any = None,
+        gt_data: str = None,
+        test_data: Union[str, list] = None,
+        metrics="similarity",
+        similarity_model_id: str = "openai/clip-vit-large-patch14",
+        resolution=(512, 512),
```

Comment:
not sure that providing resolution without resizing input image has effect for all pipelines (I know that only flux ignore input image resolution other models uses it as input resolution)

Reply:
What is your suggestion here? Remove this parameter and let the pipeline decide based on the input image or the pipeline's intrinsic parameters?

Reply:
it will depend on is behavior of all backends (diffusers, optimum, genai) aligned or not. The basic problem here is if baseline and target model have different preferences between input image resolution and manually specified height/width we may get images of different size (not sure but may be problem for embeddings comparison and required some additional postprocessing steps)
in optimum it is aligned with diffusers
@ilya-lavrenov which logic uses genai for selection image size?

Reply:
>which logic uses genai for selection image size?

based on input image size
https://github.com/openvinotoolkit/openvino.genai/blob/f177ffc9799ef34a57b257e1811a60c68c167eb2/src/cpp/src/image_generation/stable_diffusion_pipeline.hpp#L424-L431

Reply:
Ideally, the behavior should be aligned across various APIs to provide a smooth transition from one API to another.

Reply:
it's aligned with HF:
```python
                init_latents = [
                    retrieve_latents(self.vae.encode(image[i : i + 1]), generator=generator[i])
                    for i in range(batch_size)
                ]
```
https://github.com/huggingface/diffusers/blob/8ce37ab055372dedf4e9621ed63374a019d93f5d/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py#L750-L753

Here initial `image` is converted to latent space and used as initial latent.

Input `width` or `height` parameters are not used as in text to image.

VAE encode uses https://github.com/huggingface/diffusers/blob/8ce37ab055372dedf4e9621ed63374a019d93f5d/src/diffusers/image_processor.py#L540-L585

Reply:
So, I believe I should remove the resolution as a parameter and let the pipeline decide on it, right?

Reply:
for image2image - yes